safenode/src/vault/api.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::{
+    error::{Error, Result},
+    Vault,
+};
+use crate::protocol::{
+    messages::{Cmd, CmdResponse, Query, QueryResponse, ReplicatedData, Request, Response},
+    types::{address::ChunkAddress, chunk::Chunk, errors::Error as ProtocolError},
+};
+use futures::future::select_all;
+use libp2p::PeerId;
+use std::{collections::HashSet, time::Duration};
+use xor_name::XorName;
+
+impl Vault {
+    /// Store `ReplicatedData` to the closest nodes
+    pub async fn store_data(&self, data: &ReplicatedData) -> Result<()> {
+        info!("Storing data: {:?}", data.name());
+        let cmd = match data {
+            ReplicatedData::Chunk(chunk) => Cmd::StoreChunk(chunk.clone()),
+            ReplicatedData::RegisterWrite(cmd) => Cmd::Register(cmd.clone()),
+            ReplicatedData::RegisterLog(_) => todo!(),
+        };
+        // forward to the other closest nodes if we're seeing the data for the first time
+        // return early if we already have the data with us
+        match self.storage.store(&cmd).await {
+            CmdResponse::StoreChunk(Err(_)) => return Ok(()),
+            CmdResponse::CreateRegister(Err(_)) => return Ok(()),
+            CmdResponse::EditRegister(Err(_)) => return Ok(()),
```

Comment:
I think you could match them all with `CmdResponse::StoreChunk(Err(_)) |CmdResponse::CreateRegister(Err(_)) | CmdResponse::EditRegister(Err(_)) => return Ok(())`